presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardRange.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.storage.organization;
+
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestShardRange
+{
+    @Test
+    public void testEnclosingRange()
+            throws Exception
+    {
+        List<Type> types1 = ImmutableList.of(BIGINT);
+        ShardRange range1 = ShardRange.of(new Tuple(types1, 2L), new Tuple(types1, 5L));
+
+        // equal ranges enclose each other
+        assertTrue(range1.encloses(range1));
+
+        ShardRange enclosesRange1 = ShardRange.of(new Tuple(types1, 1L), new Tuple(types1, 10L));
+        ShardRange notEnclosesRange1 = ShardRange.of(new Tuple(types1, 1L), new Tuple(types1, 4L));
+
+        assertTrue(enclosesRange1.encloses(range1));
+        assertFalse(notEnclosesRange1.encloses(range1));
+
+        List<Type> types2 = ImmutableList.of(BIGINT, VARCHAR);
+        ShardRange range2 = ShardRange.of(new Tuple(types2, 2L, "aaa"), new Tuple(types2, 5L, "ccc"));
+        ShardRange enclosesRange2 = ShardRange.of(new Tuple(types2, 1L, "ccc"), new Tuple(types2, 10L, "ccc"));
+        ShardRange notEnclosesRange2 = ShardRange.of(new Tuple(types2, 2L, "aaa"), new Tuple(types2, 5L, "bbb"));
+
+        assertTrue(enclosesRange2.encloses(range2));
```

Comment:
Are `encloses` and `overlaps` supposed to be symmetric? If so, add tests for that, too